ciphers/AES.java

```diff
@@ -3,552 +3,589 @@
+    }
+
+    /**
+     * Applies the inverse Rijndael MixColumns for decryption to the input and
+     * returns the result.
+     *
+     * @param ciphertext
+     */
+    public static BigInteger mixColumnsDec(BigInteger ciphertext) {
+
+        int[] cells = splitBlockIntoCells(ciphertext);
+        int[] outputCells = new int[16];
+
+        for (int i = 0; i < 4; i++) {
+            int[] row = {cells[i * 4], cells[i * 4 + 1], cells[i * 4 + 2], cells[i * 4 + 3]};
+
+            outputCells[i * 4] = MULT14[row[0]] ^ MULT11[row[1]] ^ MULT13[row[2]] ^ MULT9[row[3]];
+            outputCells[i * 4 + 1] = MULT9[row[0]] ^ MULT14[row[1]] ^ MULT11[row[2]] ^ MULT13[row[3]];
+            outputCells[i * 4 + 2] = MULT13[row[0]] ^ MULT9[row[1]] ^ MULT14[row[2]] ^ MULT11[row[3]];
+            outputCells[i * 4 + 3] = MULT11[row[0]] ^ MULT13[row[1]] ^ MULT9[row[2]] ^ MULT14[row[3]];
+        }
+        return mergeCellsIntoBlock(outputCells);
+    }
+
+    /**
+     * Encrypts the plaintext with the key and returns the result
+     *
+     * @param plainText which we want to encrypt
+     * @param key the key for encrypt
+     * @return EncryptedText
+     */
+    public static BigInteger encrypt(BigInteger plainText, BigInteger key) {
+        BigInteger[] roundKeys = keyExpansion(key);
+
+        // Initial round
+        plainText = addRoundKey(plainText, roundKeys[0]);
+
+        // Main rounds
+        for (int i = 1; i < 10; i++) {
+            plainText = subBytes(plainText);
+            plainText = shiftRows(plainText);
+            plainText = mixColumns(plainText);
+            plainText = addRoundKey(plainText, roundKeys[i]);
+        }
+
+        // Final round
+        plainText = subBytes(plainText);
+        plainText = shiftRows(plainText);
+        plainText = addRoundKey(plainText, roundKeys[10]);
+
+        return plainText;
+    }
+
+    /**
+     * Decrypts the ciphertext with the key and returns the result
+     *
+     * @param cipherText The Encrypted text which we want to decrypt
+     * @param key
+     * @return decryptedText
+     */
+    public static BigInteger decrypt(BigInteger cipherText, BigInteger key) {
+
+        BigInteger[] roundKeys = keyExpansion(key);
+
+        // Invert final round
+        cipherText = addRoundKey(cipherText, roundKeys[10]);
+        cipherText = shiftRowsDec(cipherText);
+        cipherText = subBytesDec(cipherText);
+
+        // Invert main rounds
+        for (int i = 9; i > 0; i--) {
+            cipherText = addRoundKey(cipherText, roundKeys[i]);
+            cipherText = mixColumnsDec(cipherText);
+            cipherText = shiftRowsDec(cipherText);
+            cipherText = subBytesDec(cipherText);
+        }
+
+        // Invert initial round
+        cipherText = addRoundKey(cipherText, roundKeys[0]);
+
+        return cipherText;
+    }
+
+    public static void main(String[] args) {
+
+        try (Scanner input = new Scanner(System.in)) {
```

Comment:
Please change this construct. 
```java
try {
// somehting

}
catch (<Exception> e) {
// somehting
}
```

Reply:
Why ?
this is try block with resources and it's available in Java from JDK8 (Java 8)
this is my references about it from Oracle Java docs 
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
if you still want it a normal try-catch block then I'll do it.